services/worker/src/worker/job_runners/config/parquet_metadata.py

```diff
@@ -26,17 +28,34 @@
     ParquetFileMetadataItem,
 )
 from worker.job_runners.config.config_job_runner import ConfigJobRunner
-from worker.utils import get_parquet_file
+from worker.utils import hf_hub_open_file, hffs_parquet_url
 
 SLEEPS = [0.2, 1, 1, 10, 10, 10]
 
 
+# TODO: on which error?
+@retry(sleeps=SLEEPS)
+def retry_open_file(
+    file_url: str, hf_endpoint: str, hf_token: Optional[str], revision: Optional[str] = None
+) -> HfFileSystemFile:
+    return hf_hub_open_file(file_url=file_url, hf_endpoint=hf_endpoint, hf_token=hf_token, revision=revision)
+
+
 def create_parquet_metadata_file_from_remote_parquet(
-    parquet_file_item: SplitHubFile, fs: HTTPFileSystem, hf_token: Optional[str], parquet_metadata_directory: StrPath
+    parquet_file_item: SplitHubFile, hf_endpoint: str, hf_token: Optional[str], parquet_metadata_directory: StrPath
 ) -> ParquetFileMetadataItem:
+    split_directory = extract_split_name_from_parquet_url(parquet_file_item["url"])
```

Comment:
maybe we can rename `extract_split_name_from_parquet_url` -> `extract_split_directory_from_parquet_url ` since the split directory is different from the split name: it can be e.g. `train` or `partial-train` or `train-part0`